rtsp.go

```diff
@@ -164,47 +194,55 @@ func (rc *rtspCamera) reconnectClient() (err error) {
 		return err
 	}
 
-	// setup RTP/H264 -> H264 decoder
 	rtpDec, err := format.CreateDecoder2()
 	if err != nil {
-		return err
+		rc.logger.Errorf("error creating H264 decoder %v", err)
 	}
 
-	// setup H264 -> raw frames decoder
-	rc.rawDecoder, err = newH264Decoder()
+	rc.rawDecoder, err = newDecoder(C.AV_CODEC_ID_H264)
 	if err != nil {
 		return err
 	}
 
-	// if SPS and PPS are present into the SDP, send them to the decoder
+	// For H.264, handle SPS and PPS (assuming your H264 format has these fields)
 	if format.SPS != nil {
 		rc.rawDecoder.decode(format.SPS)
+	} else {
+		rc.logger.Warnf("no SPS found in H264 format")
```

Comment:
```suggestion
		rc.logger.Warn("no SPS found in H264 format")
```
for log statement with no output to print other than your string message